### ISSUE TITLE
`multiline-comment-style`: add option to allow mutliline line comments when `starred-block` or `bare-block` is set

### ISSUE THREAD

**Issue by golopot** [eslint/eslint#11863]
**Use case:**
I want to use this rule to enforce style in star comments, while still allowing multiline line comments.

The following example shows a case where enforcing starred-block comment is doing harm.
```js
function f() {
  
  // comment 1
  if (a) {}

  /*
   * comment 2
   * The style of this comment is inconsistent to others.
   */
  if (b) {}

  // comment 3
  if (b) {}

  // comment 4
  // it is better to use line comments here.
  if (c) {}
}
```
**What rule do you want to change?**
multiline-comment-style
**Does this change cause the rule to produce more or fewer warnings?**
fewer
**How will the change be implemented? (New option, new default behavior, etc.)?**
New option schema: 
```ts
type OptionSchema = [
    | 'starred-block'
    | 'bare-block'
    | 'separate-lines'
    | {
        style: 'starred-block' | 'bare-block' | 'separate-lines';
        allowsSeparateLines: boolean;
      },
];
```
**Please provide some example code that this change will affect:**

Code will be valid with option `{style: "starred-block", "allowsSeparateLines": true}`.
```js
// a
// b
```

**Are you willing to submit a pull request to implement this change?**
Maybe.

**Comment by kaicataldo** [comment]
Adding an option like this feels like it goes against the goal of the rule, given that the intended use is to enforce one specific style for multi-line comments. Is there a reason you wouldn't just turn it off? With this option on, all the rule would be enforcing is the that multi-line block comments have or don't have asterisks.

On a different note, it seems like `{ style: "separate-lines", allowsSeparateLines: false }` is a possible configuration in this proposal. Mind clarifying what behavior you would expect?

**Comment by golopot** [comment]
> Is there a reason you wouldn't just turn it off?

Because I still need the stlying stars part the rule. And I am dismayed the functionality is bundled selled with the forbidding multiline line comment feature which I disliked.

I suppose the stlying star comment part is an generally accepted best practice. While the enforcing mulitiline comment to be star comment is controversial. For this reason many style configs will not include this rule. 

Regarding to your last paragraph, I admit the option schema is not ideal, and I have no solution.

**Comment by platinumazure** [comment]
@kaicataldo Did the [most recent reply](https://github.com/eslint/eslint/issues/11863#issuecomment-505411195) address your concerns?
